query/components_cache.go

```diff
@@ -24,25 +25,41 @@ const cacheJobBatchSize = 1000
 // <id> -> models.Component
 var componentCache = cache.New[models.Component](gocache_store.NewGoCache(gocache.New(10*time.Minute, 10*time.Minute)))
 
+// <name/type/parentID/topologyID> -> models.Component
+var componentNameTypeParentTopologyCache = cache.New[models.Component](gocache_store.NewGoCache(gocache.New(30*time.Minute, 30*time.Minute)))
+
 func componentCacheKey(id string) string {
 	return "componentID:" + id
 }
 
+func componentNameTypeParentTopologyCacheKey(name, typ string, parentID *uuid.UUID, topologyID string) string {
+	pID := uuid.Nil.String()
+	if parentID != nil {
+		pID = parentID.String()
+	}
+	return name + typ + pID + topologyID
+}
+
 func SyncComponentCache(ctx context.Context) error {
-	var next string
+	next := uuid.Nil.String()
```

Comment:
Without this, the query was always failing with `invalid input syntax for type uuid: ""`